crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -798,4 +835,66 @@ where
 
         Ok(last_segment_headers)
     }
+
+    fn subscribe_archived_object_mappings(&self, pending: PendingSubscriptionSink) {
```

Comment:
This method is problematic because it will break RPC connection unexpectedly in case there are too many object mappings produced in a segment. Given unbounded nature of this particular data structure, extra are needs to be taken to allow receiver to get all of the mappings without destroying the connection.

Reply:
What is the specific concern here?
- [RPC_DEFAULT_MAX_RESPONSE_SIZE_MB](https://github.com/autonomys/subspace/blob/b24995d11dabef39979b774d34907c8a739138ce/crates/subspace-service/src/config.rs#L32) = 15 MB
- subscription response [DEFAULT_BUF_SIZE](https://github.com/autonomys/polkadot-sdk/blob/819a5818284a96a5a5bd65ce67e69bab860d4534/substrate/client/rpc/src/utils.rs#L33) = 16 items
- something else?

If it’s just the buffer size, then I can change the subscription so it returns one response per segment, with a list of mappings for that segment. How often are segments usually produced?

Reply:
The primary concern is unbounded nature of it. Since object mappings are produced at the end of the segment (though they actually don't have to be and we can update API to expose them incrementally) and each block can have potentially hundreds of them, spitting out a huge number of them is likely to reach the limits and either dropped or cause the whole connection to break.

I have not actually experimented with it to see what will happen, but I do see this as a likely issue. Especially considering [how annoying breakage of RPC is sometimes](https://github.com/paritytech/jsonrpsee/issues/1409).

Reply:
I implemented a partial fix in [7f071bb](https://github.com/autonomys/subspace/pull/2979/commits/7f071bbf09517322a1e903fe3f8f9d270eb039af), put a TODO in the code, and added a checklist item in the notion doc.

Is there a specific way you want to limit response numbers/sizes?

I was thinking of doing some testing first, and seeing what kinds of issues happen in practice. Maybe forcing full archiving will help with these tests.

Reply:
It is almost like we need to provide some kind of feedback mechanism, but definitely doing some testing first will better inform decisions here.